openedx_caliper_tracking/tasks.py

```diff
@@ -38,8 +38,20 @@ def deliver_caliper_event_to_kafka(self, transformed_event, event_type):
         LOGGER.info('Attempt # {} of sending event: {} to kafka ({}) is in progress.'.format(
                     self.request_stack().get('retries'), event_type, _get_kafka_setting('END_POINT')))
 
-        producer = KafkaProducer(bootstrap_servers=_get_kafka_setting('END_POINT'),
-                                 value_serializer=lambda v: json.dumps(v).encode('utf-8'))
+        if _get_kafka_setting('USERNAME') and _get_kafka_setting('PASSWORD'):
+            producer = KafkaProducer(
```

Comment:
Just a thought.  What if we add some `SASL` specific config variable?

Reply:
Done